enterprise/tasks.py

```diff
@@ -8,25 +8,60 @@
 from celery import shared_task
 from edx_django_utils.monitoring import set_code_owner_attribute
 
+from django.apps import apps
+from django.core import mail
 from django.db import IntegrityError
 
-from enterprise.models import EnterpriseCourseEnrollment, EnterpriseCustomerUser, EnterpriseEnrollmentSource
+from enterprise.utils import send_email_notification_message
 
 LOGGER = getLogger(__name__)
 
 
+@shared_task
+@set_code_owner_attribute
+def send_enterprise_email_notification(
+    enterprise_customer_uuid,
+    admin_enrollment,
+    email_items,
+):
+    """
+    Send enrollment email notifications to specified learners
+
+    Arguments:
+        * enterprise_customer_uuid (UUID)
+        * admin_enrollment=False : If True, this indicates admin based enrollment (e.g., bulk enrollment)
+        *
+        * email_items: list of dictionary objects with keys:
+        *   user (dict): a dict with either of the following forms:
+              - 1: { 'first_name': name, 'username': user_name, 'email': email } (similar to a User object)
+              - 2: { 'user_email' : user_email } (similar to a PendingEnterpriseCustomerUser object)
+        *   enrolled_in (dict): name and optionally other keys needed by templates
+        *   dashboard_url (str)
+    """
+    with mail.get_connection() as email_conn:
+        for item in email_items:
```

Comment:
Since this isn't wrapped with a try/catch, I believe that an error in one email send will prevent all from being sent. Is this behavior OK for your use case?

Reply:
probably not ok. May as well fix that here, with at least a logged exception, it existed in the prior version but that doesn't mean it wasn't a problem.  For the beyond MVP stuff we want to move this to Braze, so I wouldn't try to provide failed send feedback here.

Reply:
Sorry I had merged this by the time I read this. But yeah agree with Brian's observation. I can create a separate PR to do that. And I will likely merge and test that version tomorrow, which should make this sturdier

Reply:
I don't know exactly what Exceptions are going to be thrown by the underlying code yet though.. a quick look at the mail.send() api reveals nothing. will a `except Exception be` the safest option?

Reply:
That hammer will catch it all, sounds good to me.

Reply:
to close the loop this update is now being released via https://github.com/edx/edx-platform/pull/28324